### PR TITLE
Annotate fastas remove bgzip step

### PR DESCRIPTION
**Pull request by lynnjo**
## Description

_Provide a summary of your changes including motivation and context.
If these changes fix a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [x] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated relevant documentation

## Changelog entry

_Please add a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Annotate fasta no longer compresses the new file as anchorwave needs uncompressed fastas
```

### CODE REVIEW

**Review by zrm22 — APPROVED** [review]

File: src/main/kotlin/net/maizegenetics/phgv2/agc/AnnotateFastas.kt [hunk 1]
```diff
@@ -125,14 +126,10 @@ class AnnotateFastas : CliktCommand(help = "Annotate FASTA files with sample nam
...
 
-                // if file was previously compressed, we compress the new file.
-                // Strip off the .gz extension before writing the new non-compressed file.
-                // it wil be added when we re-compress the file.
-                // The file names will either be <sampleName>.fa or <sampleName>.gz
-                // When the file is compressed, we only add .gz as an extension as AGC
-                // takes everything before the last period as the sample name and we
-                // want the sample name to look like "B73" or "CML247" and not "B73.fa" or "CML247.fa"
-                val justName = if (fastaFile.endsWith(".gz")) sampleName else "${sampleName}.fa"
+                // Previously if the original file was previously compressed, we compressed the new file.
```
Comment (tcasstevens) [inline src/main/kotlin/net/maizegenetics/phgv2/agc/AnnotateFastas.kt:129]: Should this say wasn't?

File: src/main/kotlin/net/maizegenetics/phgv2/agc/AnnotateFastas.kt [hunk 2]
```diff
@@ -125,14 +126,10 @@ class AnnotateFastas : CliktCommand(help = "Annotate FASTA files with sample nam
...
+                // We no longer do this, as the align-assemblies step requires uncompressed fasta files.
+                // The file names will  be <sampleName>.fa
+                val justName =  "${sampleName}.fa"
```
Comment (tcasstevens) [inline src/main/kotlin/net/maizegenetics/phgv2/agc/AnnotateFastas.kt:132]: What if the incoming file is already compress?

**Review by tcasstevens — APPROVED** [review]

File: src/main/kotlin/net/maizegenetics/phgv2/agc/AnnotateFastas.kt (hunk 1 above)
Comment (lynnjo) [reply src/main/kotlin/net/maizegenetics/phgv2/agc/AnnotateFastas.kt:129]: No, it is correct other than I should remove one of the "previously".

In the original code:  if the original fasta file was compressed, I compressed the annotated fasta.  If the original was uncompressed, I did not compress the created file.  

Now I am never compressing the annotated files as these files are input for align-assemblies and anchorwave does not handle compressed files.

File: src/main/kotlin/net/maizegenetics/phgv2/agc/AnnotateFastas.kt (hunk 2 above)
Comment (lynnjo) [reply src/main/kotlin/net/maizegenetics/phgv2/agc/AnnotateFastas.kt:132]: This code will never compress the output fastas as they are used in the align-assemblies step, and anchorwave does not handle compressed files.  I didn't think of that when I originally wrote the code - I was only thinking of AGC which can handle the compressed files.

**Review by pjbradbury — APPROVED** [review]

**Pull request merged**